### ISSUE TITLE
Use of gmake on FreeBSD breaks YJIT builds in MRI 3.3

### ISSUE THREAD

**Issue by Freaky** [rbenv/ruby-build#2262]
### Steps to reproduce the behavior

Install ruby-3.3.0-preview2 on FreeBSD with YJIT enabled.

### Logs

```
building Rust YJIT (release mode)
touch yjit/target/release/libyjit.a
partial linking yjit/target/release/libyjit.a into yjit/target/release/libyjit.o
false
gmake: *** [yjit/yjit.mk:49: yjit/target/release/libyjit.o] Error 1
```

Examining [`yjit.mk`](https://github.com/ruby/ruby/blob/10379ebf90acd9622e4fccd9d4386ab3a69f4e9c/yjit/yjit.mk#L44C1-L45C45):

```
# Note, BSD handling is in yjit/not_gmake.mk
```

ruby-build has [this section](https://github.com/rbenv/ruby-build/blob/27a5309660b51bca8a28871813b9d95a76abcb29/bin/ruby-build#L1360-L1366) to select gmake on FreeBSD, referencing a 4 year old bug in Ruby 2.6.5:

```
    # Workaround for Ruby bug 16331: https://bugs.ruby-lang.org/issues/16331
    # Due to this bug, build will fail with FreeBSD's make after #1368
    # The bug is already fixed in upstream but GNU make is still required
    # when building older releases of Ruby. Use GNU make rather than switching
    # depending of Ruby version.
    export MAKE="gmake"
```

Removing this results in a successful build.  Given this seems fixed in all supported versions of MRI - and BSD make is used by the lang/ruby\* ports - it seems a reasonable fix, but anyone still depending on older versions might disagree...

**Comment by mislav** [comment]
You're right that the Ruby build bug that was solved by defaulting to `gmake` on FreeBSD is old in that it only affected Ruby versions 2.6 and 2.7 at the time. Since those Ruby versions are EOL now, and the `gmake` default is now an impediment to installing Ruby 3.3, I vote that we remove the workaround. Would you submit a PR?

BSD users who need to install older Ruby versions can set `MAKE=gmake` themselves as necessary.

Ref. https://github.com/rbenv/ruby-build/pull/1381

**Comment by Freaky** [comment]
PR: https://github.com/rbenv/ruby-build/pull/2263
